src/Chapter2.hs

```diff
@@ -726,8 +757,12 @@ value of the element itself
 
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
+
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat (map (\x -> replicate x x ) l)
+
+smartReplicateConcatMap :: [Int] -> [Int]
+smartReplicateConcatMap l = concatMap (\x -> replicate x x) l
```

Comment:
Nice usage of standard functions! You can also notice an opportunity for eta-reduction here